torch/functional.py

```diff
@@ -72,6 +72,38 @@ def broadcast_tensors(*tensors):
     return _VF.broadcast_tensors(tensors)  # type: ignore
 
 
+def broadcast_shapes(*shapes):
+    r"""broadcast_shapes(*shapes) -> Size
+
+    Similar to :func:`broadcast_tensors` but for shapes.
+
+    This is equivalent to
+    `torch.broadcast_tensors(*map(torch.empty, shapes))[0].shape`
+    but avoids the need create to intermediate tensors. This is useful for
+    broadcasting tensors of common batch shape but different rightmost shape,
+    e.g. to broadcast mean vectors with covariance matrices.
+
+    Example::
+
+        >>> torch.broadcast_shapes((2,), (3, 1), (1, 1, 1))
+        torch.Size([1, 3, 2])
+
+    Args:
+        \*shapes (torch.Size): Shapes of tensors.
+
+    Returns:
+        shape (torch.Size): A shape compatible with all input shapes.
+
+    Raises:
+        RuntimeError: If shapes are incompatible.
+    """
+    # TODO Consider moving this to C++ once the jit has better support for torch.Size.
+    scalar = torch.zeros((), device="cpu")
```

Comment:
I am not sure but maybe it is better to add here `with torch.no_grad()` so that no extra autograd tracing is performed?
Also, looking at original `broadcast_shapes`, there's additional `handle_torch_function` call when scripting is disabled. I am not sure if there's some kind of function handler for `torch.Size` so probably the only check that is needed is that arguments are of type `torch.Size`. On the other hand, tuples of ints are ok arguments too, so I'm hesitating here. What do you think?

Reply:
Sure, I've added `torch.no_grad()`.

I don't understand the purpose of the `handle_torch_function`. However jit usage should be exercised in the jit tests of `MultivariateNormal`, so I guess we can see if all tests pass?

Reply:
I think distribution tests use `jit.trace` so `jit.script` is uncovered. However it means that distributions might be not working with `jit.script` aside from this PR :)